### PR TITLE
Replace simple-overlay with dialog-fullscreen component

### PR DESCRIPTION
**Pull request by annabelsegalld2l**
The simple-overlay component is being retired and replaced with the d2l-dialog-fullscreen component. It should not affect the functionality of the dashboard editor.

Here's a GIF of the new dialog: 

insight_dashboards

### CODE REVIEW

File: src/d2l-dashboard-editor.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ $_documentContainer.innerHTML = `<dom-module id="d2l-dashboard-editor">
 			}
 		</style>
 
-		<d2l-simple-overlay id="dashboard-edit-overlay" title-name="[[localize('dashboardEditorTitle')]]" close-simple-overlay-alt-text="close">
```
Comment (annabelsegalld2l) [inline src/d2l-dashboard-editor.js:43]: The close-simple-overlay-alt-text attribute is not needed, as the dialog has a built-in, localized alt-text.

File: src/d2l-dashboard-editor.js [hunk 2]
```diff
@@ -136,8 +138,8 @@ Polymer({
 		return new Boolean(bool).toString();
 	},
 	_closeEditDashboardView: function() {
-		var overlay = dom(this.root).querySelector('#dashboard-edit-overlay');
-		overlay.close();
+		var dialog = dom(this.root).querySelector('#dashboard-edit-dialog');
```
Comment (annabelsegalld2l) [inline src/d2l-dashboard-editor.js:141]: Just changed the id here from overlay to dialog to align it with the new type.

**Review by NicholasHallman — APPROVED** [review]
lgtm!

File: src/d2l-dashboard-editor.js (hunk 2 above)
Comment (svanherk) [reply src/d2l-dashboard-editor.js:141]: Nice!

**Review by svanherk — APPROVED** [review]

**Pull request merged**